pkg/controller/useraccount/useraccount_controller_test.go

```diff
@@ -306,10 +328,67 @@ func TestReconcile(t *testing.T) {
 		})
 	})
 
+	t.Run("create nstmplset OK", func(t *testing.T) {
+		t.Run("create", func(t *testing.T) {
+			r, req, _ := prepareReconcile(t, username, userAcc, preexistingUser, preexistingIdentity)
+
+			// test
+			_, err := r.Reconcile(req)
+
+			require.NoError(t, err)
+			checkStatus(t, r.client, username, "Provisioning", "")
+			checkNSTmplSet(t, r.client, username)
+		})
+
+		t.Run("update_status", func(t *testing.T) {
+			preexistingNsTmplSetWithNS := &toolchainv1alpha1.NSTemplateSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      userAcc.Name,
+					Namespace: "toolchain-member",
+				},
+				Spec: newNSTmplSetSpec(),
+				Status: toolchainv1alpha1.NSTemplateSetStatus{
+					Conditions: []toolchainv1alpha1.Condition{
+						{
+							Type:   toolchainv1alpha1.ConditionReady,
+							Status: corev1.ConditionFalse,
+							Reason: "ProvisioningNamespace",
+						},
+					},
+				},
+			}
+
+			r, req, _ := prepareReconcile(t, username, userAcc, preexistingUser, preexistingIdentity, preexistingNsTmplSetWithNS)
+
+			// test
+			_, err := r.Reconcile(req)
+
+			require.NoError(t, err)
+			checkStatus(t, r.client, username, "ProvisioningNamespace", "")
```

Comment:
this should have been already gone - we set only one `Provisioning` reason. could you please replace it for example by an error case?

Reply:
Outdated comment.